zookeeper/source/zk_nn_client.cc

```diff
@@ -369,19 +370,45 @@ bool ZkNnClient::add_block(AddBlockRequestProto &req,
 
   std::uint64_t block_id;
   auto data_nodes = std::vector<std::string>();
+  uint64_t block_group_id;
+  std::vector<char> block_indices;
 
   if (!znode_data.isEC) {
       add_block(file_path, block_id, data_nodes, replication_factor);
   } else {  // case when some EC policy is used.
-      // TODO(Nate): generate a block group and each part of a block group.
-      // TODO(Nate): use the hierarchical naming scheme.
+      add_block_group(
+          file_path, block_group_id, data_nodes, block_indices,
+          DEFAULT_PARITY_UNITS + DEFAULT_DATA_UNITS);
   }
 
+  // TODO(nate): this might be seriously wrong.
```

Comment:
We should discuss this as a team on Wednesday before merging then.

Reply:
addressed.